File: app/src/main/java/in/woloo/www/subscribe/models/PlanResponse.java
package in.woloo.www.subscribe.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class PlanResponse {

    @SerializedName("code")
    @Expose
    private Integer code;
    @SerializedName("status")
    @Expose
    private String status;
    @SerializedName("message")
    @Expose
    private String message;
    @SerializedName("data")
    @Expose
    private List<Data> data = null;

    public Integer getCode() {
        return code;
    }

    public void setCode(Integer code) {
        this.code = code;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public List<Data> getData() {
        return data;
    }

    public void setData(List<Data> data) {
        this.data = data;
    }

    public class Data{
        @SerializedName("id")
        @Expose
        private Integer id;
        @SerializedName("name")
        @Expose
        private String name;
        @SerializedName("description")
        @Expose
        private String description;
        @SerializedName("frequency")
        @Expose
        private String frequency;
        @SerializedName("days")
        @Expose
        private Integer days;
        @SerializedName("image")
        @Expose
        private String image;
        @SerializedName("price")
        @Expose
        private String price;
        @SerializedName("discount")
        @Expose
        private Object discount;
        @SerializedName("is_expired")
        @Expose
        private Integer isExpired;
        @SerializedName("status")
        @Expose
        private Integer status;
        @SerializedName("created_at")
        @Expose
        private String createdAt;
        @SerializedName("updated_at")
        @Expose
        private String updatedAt;
        @SerializedName("deleted_at")
        @Expose
        private Object deletedAt;
        @SerializedName("plan_id")
        @Expose
        private String planId;
        @SerializedName("currency")
        @Expose
        private String currency;
        @SerializedName("backgroud_color")
        @Expose
        private String backgroud_color;
        @SerializedName("shield_color")
        @Expose
        private String shield_color;
        @SerializedName("strike_out_price")
        @Expose
        private Integer beforeDiscountPrice;

        public String getBackgroud_color() {
            return backgroud_color;
        }

        public void setBackgroud_color(String backgroud_color) {
            this.backgroud_color = backgroud_color;
        }

        public String getShield_color() {
            return shield_color;
        }

        public void setShield_color(String shield_color) {
            this.shield_color = shield_color;
        }

        public Integer getId() {
            return id;
        }

        public void setId(Integer id) {
            this.id = id;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getDescription() {
            return description;
        }

        public void setDescription(String description) {
            this.description = description;
        }

        public String getFrequency() {
            return frequency;
        }

        public void setFrequency(String frequency) {
            this.frequency = frequency;
        }

        public Integer getDays() {
            return days;
        }

        public void setDays(Integer days) {
            this.days = days;
        }

        public String getImage() {
            return image;
        }

        public void setImage(String image) {
            this.image = image;
        }

        public String getPrice() {
            return price;
        }

        public void setPrice(String price) {
            this.price = price;
        }

        public Object getDiscount() {
            return discount;
        }

        public void setDiscount(Object discount) {
            this.discount = discount;
        }

        public Integer getIsExpired() {
            return isExpired;
        }

        public void setIsExpired(Integer isExpired) {
            this.isExpired = isExpired;
        }

        public Integer getStatus() {
            return status;
        }

        public void setStatus(Integer status) {
            this.status = status;
        }

        public String getCreatedAt() {
            return createdAt;
        }

        public void setCreatedAt(String createdAt) {
            this.createdAt = createdAt;
        }

        public String getUpdatedAt() {
            return updatedAt;
        }

        public void setUpdatedAt(String updatedAt) {
            this.updatedAt = updatedAt;
        }

        public Object getDeletedAt() {
            return deletedAt;
        }

        public void setDeletedAt(Object deletedAt) {
            this.deletedAt = deletedAt;
        }

        public String getPlanId() {
            return planId;
        }

        public void setPlanId(String planId) {
            this.planId = planId;
        }

        public String getCurrency() {
            return currency;
        }

        public void setCurrency(String currency) {
            this.currency = currency;
        }

        public Integer getBeforeDiscountPrice() {
            return beforeDiscountPrice;
        }

        public void setBeforeDiscountPrice(Integer beforeDiscountPrice) {
            this.beforeDiscountPrice = beforeDiscountPrice;
        }
    }

}

File: app/src/main/java/in/woloo/www/more/subscribe/models/PlanResponse.kt
package `in`.woloo.www.more.subscribe.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class PlanResponse {
    @SerializedName("code")
    @Expose
    var code: Int? = null

    @SerializedName("status")
    @Expose
    var status: String? = null

    @SerializedName("message")
    @Expose
    var message: String? = null

    @SerializedName("data")
    @Expose
    var data: List<Data>? = null

    inner class Data {
        @JvmField
        @SerializedName("id")
        @Expose
        var id: Int? = null

        @JvmField
        @SerializedName("name")
        @Expose
        var name: String? = null

        @JvmField
        @SerializedName("description")
        @Expose
        var description: String? = null

        @JvmField
        @SerializedName("frequency")
        @Expose
        var frequency: String? = null

        @JvmField
        @SerializedName("days")
        @Expose
        var days: Int? = null

        @SerializedName("image")
        @Expose
        var image: String? = null

        @JvmField
        @SerializedName("price")
        @Expose
        var price: String? = null

        @SerializedName("discount")
        @Expose
        var discount: Any? = null

        @SerializedName("is_expired")
        @Expose
        var isExpired: Int? = null

        @SerializedName("status")
        @Expose
        var status: Int? = null

        @SerializedName("created_at")
        @Expose
        var createdAt: String? = null

        @SerializedName("updated_at")
        @Expose
        var updatedAt: String? = null

        @SerializedName("deleted_at")
        @Expose
        var deletedAt: Any? = null

        @JvmField
        @SerializedName("plan_id")
        @Expose
        var planId: String? = null

        @JvmField
        @SerializedName("currency")
        @Expose
        var currency: String? = null

        @SerializedName("backgroud_color")
        @Expose
        var backgroud_color: String? = null

        @SerializedName("shield_color")
        @Expose
        var shield_color: String? = null

        @JvmField
        @SerializedName("strike_out_price")
        @Expose
        var beforeDiscountPrice: Int? = null
    }
}
